financepayment/controllers/front/response.php

```diff
@@ -640,4 +632,146 @@ protected function createSignature($payload, $secret)
 
         return $signature;
     }
+
+    /**
+     * checks the integrity and validity of the webhook payload
+     *
+     * @param string $input the raw string of the body
+     * @return object the json decoded webhook payload object
+     * @throws WebhookException if json could not be decoded
+     * @throws WebhookException if secret set but no HMAC received in the header
+     * @throws WebhookException if divido HMAC does not match HMAC generated with config shared secret
+     * @throws WebhookException if a required element of the payload is missing
+     */
+    protected function validateWebhook($input){
+        $data  = json_decode($input);
+        if(!$data){
+            throw new WebhookException("Could not decode json payload of request", self::HTTP_RESPONSE_CODE_BAD_REQUEST);
+        }
+
+        // only check HMAC if a secret is configured in the plugin config
+        if (!empty(Configuration::get('FINANCE_HMAC'))){
+            $callback_sign = isset($_SERVER['HTTP_X_DIVIDO_HMAC_SHA256']) ?  $_SERVER['HTTP_X_DIVIDO_HMAC_SHA256']  : null;
+            if(empty($callback_sign)){
```

Comment:
You could use the [null coalescing operator](https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op) here, e.g.

```suggestion
            if(empty($_SERVER['HTTP_X_DIVIDO_HMAC_SHA256'] ?? null)) {